source/setup.py

```diff
@@ -175,7 +176,8 @@ def _genManifestTemplate(shouldHaveUIAccess: bool) -> tuple[int, int, bytes]:
 	],
 	options={
 		"verbose": 2,
-		"optimize": 0,
+		# Only raise errors on assertions for test builds.
+		"optimize": 0 if isTestVersion else 1,

```

Comment:
Note that this approach is a logic change from  what was done for many years, i.e. the code was optimized when providing `release` to the `scons` invocation. This offered us two advantages gone with the current implementation:
1. Beta builds were optimized ensuring any issues caused by Python code being compiled with the `-o` flag were discovered before RC stage (for an example where this was useful see#13508)
2.  It was possible to create a try branch identical to a release by naming it in a particular way - this is once again impossible with the current design

Given @michaelDCurran approved this I assume you have discussed the change internally and reached the conclusion that these aspects aren't important - in that case the PR description should be updated to explain why.
If this was an oversight I'd suggest to pass the `release` parameter as an CLI argument to the script and use it to determine the value of optimization.

Reply:
@michaelDCurran - is this a more appropriate solution?

```suggestion
		"optimize": 0 if "alpha" in version else 1,
```